AEPServices/Sources/log/DefaultStringInterpolation+EventDataPrinter.swift

```diff
@@ -0,0 +1,25 @@
+/*
+ Copyright 2021 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+
+public extension DefaultStringInterpolation {
```

Comment:
Looks good, I'm a bit confused where this is used by the logging service? Also, I'd prefer if this wasn't a public extension so we aren't exposing to customers if possible.

Reply:
Appears that this affects how all interpolated strings are printed in swift, even outside the SDK. I'm not sure we can go this route as it could potentially affect customer's logging and other string manipulations.

Reply:
The logging service just simply accepts any string as an argument and print it, so what we need to do is to provide a prettified `Event Data` JSON string by using either a utility function or in this PR using a `string interpolation` function .
In fact, the approach of enhancing `string interpolation` do have the side effect which affects customer's String manipulations. But the customer will not run into this code if they don't use it in their code.
```Swift
"\(prettyEventData:["key":"value"])"
```
Either way is ok with me, @nporter-adbe  @shalehaha @pfransenadb any comments/ideas?

Reply:
Can we make this extension internal so it won't affect customers in any way?

Reply:
We can't, if so, other AEP extensions will not have access to `prettyEventData `.

Reply:
It's not clear to me how this is intended to be used. Does the logging service use it by default or do extensions need to make changes to use this? Can you provide sample usage in the PR description?